File: rpg/schemas/memory.py
"""schemas.memory — 记忆管理路由请求模型。"""
from __future__ import annotations
from typing import Optional
from schemas._common import _BaseRequest


class MemoryModeRequest(_BaseRequest):
    mode: Optional[str] = "normal"


class MemoryAddRequest(_BaseRequest):
    bucket: Optional[str] = "notes"
    text: Optional[str] = ""


class MemoryRemoveRequest(_BaseRequest):
    bucket: Optional[str] = "notes"
    index: Optional[int] = -1

File: rust/crates/rpg-schemas/src/memory.rs
//! schemas.memory — 记忆管理路由请求模型。
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MemoryModeRequest {
    #[serde(default = "default_mode")]
    pub mode: Option<String>,
}

fn default_mode() -> Option<String> { Some("normal".to_string()) }

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MemoryAddRequest {
    #[serde(default = "default_bucket")]
    pub bucket: Option<String>,
    #[serde(default)]
    pub text: Option<String>,
}

fn default_bucket() -> Option<String> { Some("notes".to_string()) }

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MemoryRemoveRequest {
    #[serde(default = "default_bucket_remove")]
    pub bucket: Option<String>,
    #[serde(default = "default_index")]
    pub index: Option<i64>,
}

fn default_bucket_remove() -> Option<String> { Some("notes".to_string()) }
fn default_index() -> Option<i64> { Some(-1) }
